File: Src/Icm.Core/Localization/ILocalizationRepositoryExtensions.cs
using System.Globalization;
using System.Runtime.CompilerServices;

namespace Icm.Localization
{

	public static class LocalizationRepositoryExtensions
	{
		public static string Trans(this ILocalizationRepository locRepo, string key)
		{
			return locRepo[CultureInfo.CurrentCulture.LCID, key];
		}

	}

}

File: Src2/Icm.Core/Localization/ILocalizationRepositoryExtensions.vb
Imports System.Runtime.CompilerServices

Namespace Icm.Localization

    Public Module ILocalizationRepositoryExtensions

        <Extension>
        Public Function Trans(locRepo As ILocalizationRepository, key As String) As String
            Return locRepo(CultureInfo.CurrentCulture.LCID, key)
        End Function

    End Module

End Namespace